tests/components/scene/test_init.py

```diff
@@ -1,149 +1,143 @@
-        assert 100 == self.light_2.last_call("turn_on")[1].get("brightness")
-
-    def test_activate_scene(self):
-        """Test active scene."""
-        assert setup_component(
-            self.hass,
-            scene.DOMAIN,
-            {
-                "scene": [
-                    {
-                        "name": "test",
-                        "entities": {
-                            self.light_1.entity_id: "on",
-                            self.light_2.entity_id: {"state": "on", "brightness": 100},
-                        },
-                    }
-                ]
-            },
-        )
-        self.hass.block_till_done()
-
-        common.activate(self.hass, "scene.test")
-        self.hass.block_till_done()
-
-        assert self.light_1.is_on
-        assert self.light_2.is_on
-        assert self.light_2.last_call("turn_on")[1].get("brightness") == 100
-
-        turn_on_calls = mock_service(self.hass, "light", "turn_on")
-
-        self.hass.services.call(
-            scene.DOMAIN, "turn_on", {"transition": 42, "entity_id": "scene.test"}
-        )
-        self.hass.block_till_done()
-
-        assert len(turn_on_calls) == 1
-        assert turn_on_calls[0].domain == "light"
-        assert turn_on_calls[0].service == "turn_on"
-        assert turn_on_calls[0].data.get("transition") == 42
+from tests.common import async_mock_service
+
+
+@pytest.fixture(autouse=True)
+def entities(hass):
+    """Initialize the test light."""
+    platform = getattr(hass.components, "test.light")
+    platform.init()
+    yield platform.ENTITIES[0:2]
+
+
+async def test_config_yaml_alias_anchor(hass, entities):
+    """Test the usage of YAML aliases and anchors.
+
+    The following test scene configuration is equivalent to:
+
+    scene:
+      - name: test
+        entities:
+          light_1: &light_1_state
+            state: 'on'
+            brightness: 100
+          light_2: *light_1_state
+
+    When encountering a YAML alias/anchor, the PyYAML parser will use a
+    reference to the original dictionary, instead of creating a copy, so
+    care needs to be taken to not modify the original.
+    """
+    light_1, light_2 = await setup_lights(hass, entities)
+    entity_state = {"state": "on", "brightness": 100}
+
+    assert await async_setup_component(
+        hass,
+        scene.DOMAIN,
+        {
+            "scene": [
+                {
+                    "name": "test",
+                    "entities": {
+                        light_1.entity_id: entity_state,
+                        light_2.entity_id: entity_state,
+                    },
+                }
+            ]
+        },
+    )
+    await hass.async_block_till_done()
+
+    await activate(hass, "scene.test")
+
+    assert light.is_on(hass, light_1.entity_id)
+    assert light.is_on(hass, light_2.entity_id)
+    assert 100 == light_1.last_call("turn_on")[1].get("brightness")
+    assert 100 == light_2.last_call("turn_on")[1].get("brightness")
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)